lib/ex_unit/lib/ex_unit/runner.ex

```diff
@@ -90,52 +93,50 @@ defmodule ExUnit.Runner do
   # Loop expecting messages from the spawned modules. Whenever
   # a module has finished executing, decrease the taken modules
   # counter and attempt to spawn new ones.
-  defp wait_until_available(config, taken) do
+  defp wait_until_available(modules, taken, config) do
     receive do
       {_pid, :module_finished, _test_case} ->
-        loop(config, taken - 1)
+        loop(modules, taken - 1, config)
     end
   end
 
-  defp spawn_modules(config, modules, taken) do
-    pid = self()
-
+  defp spawn_modules(modules, modules_left, taken, config) do
     Enum.each(modules, fn module ->
       spawn_link(fn ->
-        run_module(config, pid, module)
+        run_module(module, config)
       end)
     end)
 
-    loop(config, taken + length(modules))
+    loop(modules_left, taken + length(modules), config)
   end
 
-  defp run_module(config, pid, module) do
+  defp run_module(module, config) do
     test_module = module.__ex_unit__()
     EM.module_started(config.manager, test_module)
 
     # Prepare tests, selecting which ones should
     # run and which ones were skipped.
-    tests = prepare_tests(config, test_module.tests)
+    tests = prepare_tests(test_module.tests, config)
```

Comment:
Please revert everywhere we change the order of arguments. It is an unnecessary change that complicates history with no benefit. We will keep `config` as first argument. Thanks.